### ISSUE TITLE
iFrame resizing with larger navbar

### ISSUE THREAD

**Issue by nik-humphries** [openanalytics/shinyproxy#341]
When using a navbar that is larger than the default one in ShinyProxy, the iFrame in the apps.html doesn't resize as expected. There is a white block of unnecessary padding that I have removed by important an apps.css, but there are also 2 scrollbars, where there should only be one.

![image](https://user-images.githubusercontent.com/20989616/143553435-3766e4a7-bcbf-4bc5-8b1f-893763926d90.png)

Previously to remove the additional scroll bar, I edited the app.html inline js (v2.5.0)
```js
<script type="text/javascript" th:inline="javascript">
		function setShinyframeHeight() {
			$('#shinyframe').css('height', ($(window).height() - $(navbar).height()) + 'px');
			$('#shinyframe').css('border', '0px');
			$('#shinyframe').css('padding', '0px');
			$('#shinyframe').css('margin', '0px');
		}
...
```
The original version (and the version in the now sourced .js script) is
```js
$('#shinyframe').css('height', ($(window).height()) + 'px')
```

The solution in v2.6.0 is to source a version of https://github.com/openanalytics/shinyproxy/blob/master/src/main/resources/static/js/shiny.ui.js that has been modified to have line 67 replaced by 
```js
$('#shinyframe').css('height', ($(window).height() - $(navbar).height()) + 'px');
```

I'm unsure if this is an issue with me using a modified navbar, but this changes the formatting so that the apps screen looks as follows
![image](https://user-images.githubusercontent.com/20989616/143567590-5e5c1407-05dc-41c3-a4c6-aa1a97265a7c.png)

It now doesn't have the padding and correctly gets the one scrollbar.

**Comment by LEDfan** [comment]
Hi

Thanks for opening this issue. You can still do this by just editing the `app.html` template using the following code:

```html
<script type="text/javascript" th:inline="javascript">
Shiny.ui.setShinyFrameHeight = function() {
	console.log('override');
	$('#shinyframe').css('height', ($(window).height() - $(navbar).height()) + 'px');
}
$(window).on('load', function() {
		  window.Shiny.common.init([[${contextPath}]], [[${application_name}]]);
	if ([[${operatorEnabled}]]) {
		window.Shiny.operator.init([[${operatorForceTransfer}]], [[${operatorShowTransferMessage}]]);
	}
	window.Shiny.app.start(
			[[${containerPath}]],
			[[${webSocketReconnectionMode}]],
			[[${proxyId}]],
			[[${heartbeatRate}]],
			[[${appName}]],
			[[${appInstance}]],
			[[${maxInstances}]],
			[[${shinyForceFullReload}]],
	);
});
</script>
```
Place this code instead of the default here: https://github.com/openanalytics/shinyproxy/blob/master/src/main/resources/templates/app.html#L117-L134
Unfortunately, this still does not work nicely with the various loading screens. Therefore I think we should try to improve the ShinyProxy code to work with a navbar of any size. I'll keep this issue open to track the progress of this enhancement.

**Comment by nik-humphries** [comment]
I'll give this a go, thanks.

**Comment by LEDfan** [comment]
Hi, today we released ShinyProxy 3.0.0 which now supports a navbar of any height. So that means that the above workaround is no longer required.
